server/cluster/cluster.go

```diff
@@ -265,7 +265,13 @@ func (s *Server) Delete(ctx context.Context, q *cluster.ClusterQuery) (*cluster.
 	if err := s.enf.EnforceErr(ctx.Value("claims"), rbacpolicy.ResourceClusters, rbacpolicy.ActionDelete, createRBACObject(c.Project, c.Server)); err != nil {
 		return nil, err
 	}
-	err = s.db.DeleteCluster(ctx, q.Server)
+
+	server := q.Server
```

Comment:
also rbac verification working only by server right now, so in case with deletion by name it will not work

Reply:
Thanks for pointing this out. Would this require further changes other than tweaking `createRBACObject`?

Reply:
It looks like the `getClusterWith403IfNotExist` does pull back the cluster info from the cluster name and could be used to pass the server info to `createRBACObject` func. Maybe there's something I'm missing?

Reply:
Looks like we will map name to server to resolve this. See changes now for further details.